include/SFML/Audio/SoundRecorder.hpp

```diff
@@ -285,13 +286,13 @@ class SFML_AUDIO_API SoundRecorder : AlResource
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    Thread             m_thread;             //!< Thread running the background recording task
-    std::vector<Int16> m_samples;            //!< Buffer to store captured samples
-    unsigned int       m_sampleRate;         //!< Sample rate
-    Time               m_processingInterval; //!< Time period between calls to onProcessSamples
-    bool               m_isCapturing;        //!< Capturing state
-    std::string        m_deviceName;         //!< Name of the audio capture device
-    unsigned int       m_channelCount;       //!< Number of recording channels
+    std::optional<std::thread> m_thread;             //!< Thread running the background recording task
```

Comment:
`std::thread` would suffice here.

I assume this `std::optional` was introduced to allow for resetting `m_thread` to a default value but `std::thread::joinable() == false` is equivalent to `std::nullopt` here.

The change for resetting would be:
```cpp
std::thread thread_1;
std::optional<std::thread> thread_2;
// Reset
thread_1 = {};
thread_2 = std::nullopt;
// Check
if (thread_1.joinable()) { /* do work */ }
if (thread_2 && thread_2->joinable()) { /* do work */ }
```

Reply:
Nice. That would also get rid of having two separate null states for `m_thread` (empty optional, non-joinable thread).

Reply:
Thanks, that should work. It's a nice improvement.